bootstrap/src/main/java/io/jenkins/jenkinsfile/runner/bootstrap/Bootstrap.java

```diff
@@ -154,7 +154,7 @@ public void postConstruct(CmdLineParser parser) throws IOException {
 
         if (showVersion) {
             System.out.println(getVersion());
-            System.exit(0);
+            Runtime.getRuntime().halt(0);
```

Comment:
this will in theory shut faster - no shutdown hooks right? will it show up as an error in any shells though?

Reply:
This will enable graceful shutdown.

I am not sure about the showing up as an error in any shells though. How can I be sure?

Reply:
I think it may be the other way around: https://www.baeldung.com/java-runtime-halt-vs-system-exit#:~:text=exit()%20method%20triggers%20the,()%20terminates%20the%20JVM%20abruptly.

if its graceful that you want, then perhaps leave it the way it was?